### PR TITLE
Clarify config recycles

### PR DESCRIPTION
**Pull request by zachmargolis**

### CODE REVIEW

File: _articles/appdev-deploy.md [hunk 1]
```diff
@@ -255,7 +255,9 @@ aws-vault exec prod-power -- ./bin/scale-remove-new-instances prod worker
 ### Config Recycle
 
 A config recycle is an abbreviated "deploy" that deploys the same code, but lets boxes pick up
-new configurations (config from S3, or service provides from `identity-idp-config`).
+new configurations (config from S3).
+
+**Note:** `identity-idp-config` changes likey likely need a migration.
```
Comment (mitchellhenke) [inline _articles/appdev-deploy.md:260]:
```suggestion
**Note:** `identity-idp-config` changes only need a migration recycle unless it includes a logo chance, in which case a full IDP recycle with migration is needed.
```

Feels wordy, but that's the current state

Comment (zachmargolis) [reply _articles/appdev-deploy.md:260]: I thought we just discovered that cert updates were being skipped over?

Comment (zachmargolis) [reply _articles/appdev-deploy.md:260]: I synced with @mitchellhenke, for now, all `identity-idp-config` changes do seem to require a migration instance so I'm going to reword this to be more explicit:

```suggestion
**Note:** currently, all `identity-idp-config` changes require a migration.
```

File: _articles/appdev-deploy.md [hunk 2]
```diff
@@ -255,7 +255,9 @@ aws-vault exec prod-power -- ./bin/scale-remove-new-instances prod worker
 ### Config Recycle
 
 A config recycle is an abbreviated "deploy" that deploys the same code, but lets boxes pick up
-new configurations (config from S3, or service provides from `identity-idp-config`).
+new configurations (config from S3).
+
+**Note:** currently, all `identity-idp-config` changes require a migration.
```
Comment (julialeague) [inline _articles/appdev-deploy.md:260]: This is probably fine for this section. I think either adding a new section here about identity-idp-config deployment process, or linking to another page about that might be useful as a future improvement. I'll check to see that a JIRA is logged. Thank you for the update, it's a helpful clarification!

**Review by julialeague — APPROVED** [review]
